### PR TITLE
Add source and target options for cancel-backfill

### PR DESCRIPTION
**Pull request by concubidated**
Addresses #25

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -116,12 +116,22 @@ has been made so far.
 				panic(errors.WithStack(err))
 			}
 
+			source, err := cmd.Flags().GetBool("source")
```
Comment (alram) [inline main.go:119]: can you check the long description above for `cancel-backfill` and see if it's worth updating as well?

File: main_test.go [hunk 2]
```diff
@@ -121,13 +121,17 @@ func TestCalcPgMappingsToUndoBackfill(t *testing.T) {
```
Comment (alram) [inline main_test.go:121]: cam you add a `source: true`, `target: true` test case as well?

Comment (concubidated) [reply main_test.go:121]: yup yup!

File: main.go (hunk 1 above)
Comment (concubidated) [reply main.go:119]: doesnt look like it needs to be updated, none of the other flags are actually described in there either, just the logic of what it does.

File: README.md [hunk 3]
```diff
@@ -108,6 +108,8 @@ $ ./pgremapper cancel-backfill [--exclude-backfilling] [--include-osds <osdspec>
 * `--include-osds`: Cancel backfills containing one of the given OSDs as a backfill source or target only.
 * `--exclude-osds`: The inverse of `--include-osds` - cancel backfills that do not contain one of the given OSDs as a backfill source or target.
 * `--pgs-including`: Cancel backfills for PGs that include the given OSDs in their up or acting set, whether or not the given OSDs are backfill sources or targets in those PGs.
+* `--source`: Used in conjunction with the above flags, only selects OSDs that are backfill sources.
```
Comment (baergj) [inline README.md:111]: s/`only selects`/`selects only`/

File: README.md [hunk 4]
```diff
@@ -108,6 +108,8 @@ $ ./pgremapper cancel-backfill [--exclude-backfilling] [--include-osds <osdspec>
 * `--include-osds`: Cancel backfills containing one of the given OSDs as a backfill source or target only.
 * `--exclude-osds`: The inverse of `--include-osds` - cancel backfills that do not contain one of the given OSDs as a backfill source or target.
 * `--pgs-including`: Cancel backfills for PGs that include the given OSDs in their up or acting set, whether or not the given OSDs are backfill sources or targets in those PGs.
+* `--source`: Used in conjunction with the above flags, only selects OSDs that are backfill sources. 
+* `--target`: Only selects OSDs that are backfill targets
```
Comment (baergj) [inline README.md:112]: ditto

File: main.go [hunk 5]
```diff
@@ -733,15 +745,27 @@ func calcPgMappingsToUndoBackfill(excludeBackfilling bool, excludedOsds, include
 							continue
 						}
 
-						if excluded(up[i]) || excluded(acting[i]) {
-							continue
-						}
-
 						// We'll allow this OSD to be
 						// acted on if this PG is the
 						// source _or_ target
```
Comment (baergj) [inline main.go:750]: This comment is a bit lost now - it should probably move into the if branch

**Review by baergj — APPROVED** [review]
Minor comment on a comment, otherwise LGTM

**Pull request merged**